tests/Extensions/Hosting.Services.Web.UnitTests/ObsoleteCorrelationHeadersMiddlewareTests.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Diagnostics;
+using System.Reflection.Metadata.Ecma335;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Omex.Extensions.Hosting.Services.Web.Middlewares;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+namespace Microsoft.Omex.Extensions.Hosting.Services.Web.UnitTests
+{
+	[TestClass]
+	public sealed class ObsoleteCorrelationHeadersMiddlewareTests
+	{
+		[Obsolete]
+		[TestInitialize]
+		public void Initialize()
+		{
+			TestActivity = new Activity("ObsoleteCorrelationHeadersMiddlewareTests");
+			TestActivity.Start();
+			TestMiddleware = new ObsoleteCorrelationHeadersMiddleware();
+			TestHttpContext = new DefaultHttpContext();
+		}
+
+		[TestCleanup]
+		public void Cleanup()
+		{
+			TestActivity.Stop();
+		}
+
+		[Obsolete]
+		[DataTestMethod]
+		[DataRow("MS-CorrelationId")]
+		[DataRow("ms-correlationid")]
+		[DataRow("corr")]
+		[DataRow("HTTP_X_CORRELATIONID")]
+		public async Task SetsObsoleteCorrelationIdFromHeaderToActivity(string correlationName)
+		{
+			// Arrange
+			Guid obsoleteCorrelation = new Guid();
```

Comment:
Using new Guid() will create a default (all 0) GUID. In order to make ensure that you don't always return the default GUID it would make sense to test using a real value here. Could be hardcoded, or as below
```suggestion
			Guid obsoleteCorrelation = Guid.NewGuid();
```